src/Framework/Framework/Controls/Validator.cs

```diff
@@ -83,7 +86,22 @@ public class Validator : HtmlGenericControl
 
         private static void AddValidatedValue(IHtmlWriter writer, IDotvvmRequestContext context, DotvvmProperty prop, DotvvmControl control)
         {
-            writer.AddKnockoutDataBind("dotvvm-validation", control, ValueProperty, renderEvenInServerRenderingMode: true);
+            const string validationDataBindName = "dotvvm-validation";
+
+            var binding = control.GetValueBinding(ValueProperty);
+            if (binding is not null)
+            {
+                var referencedPropertyExpressions = binding.GetProperty<ReferencedViewModelPropertiesBindingProperty>();
+                var unwrappedPropertyExpression = referencedPropertyExpressions.UnwrappedBindingExpression;
+
+                // We were able to unwrap the the provided expression
+                writer.AddKnockoutDataBind(validationDataBindName, control, unwrappedPropertyExpression, renderEvenInServerRenderingMode: true);
+            }
+            else
+            {
+                // We are passing the binding "as-is"
+                writer.AddKnockoutDataBind(validationDataBindName, control, ValueProperty, renderEvenInServerRenderingMode: true);
```

Comment:
I think it won't really work without the value binding, right? I'd just throw an exception here, unless there exists a scenario when validation without Validator.Value works...